### DISCUSSION TITLE
While uploading IPA to App Store Connect got this error: "*** -[__NSArrayM insertObject:atIndex:]: object cannot be nil (-1010)"

### DISCUSSION THREAD

**Discussion by UraFIZ** [fastlane/fastlane#21262] [General]
Mac: 13.2.1 Fastlane: 2.212.2

I upload my project using the command "fastlane ios upload_build." I managed to deliver IPA to App Store Connect several times. Still, after couple more times, I got this error: "Exception caught. *** -[__NSArrayM insertObject:atIndex:]: object cannot be nil (-1010) The call to the altool completed with non-zero exit status: 1. This indicates a failure."

What could be the reason for this error?

**Comment by fespinoza** [comment]
I saw this in the apple forums https://developer.apple.com/forums/thread/729645

in my own project I realized I had issues with my version number in the Xcode project due to a wrong tag name, so the version number of the build was not set correctly

**Reply by janis-ax** [reply]
Agree, I had exactly this issue

**Reply by asbjornu** [reply]
Thanks! It would be helpful if `increment_version_number` validated the version number and provided a more user-friendly error message. :)

**Reply by williamdes** [reply]
I reported this with more detail in: https://github.com/fastlane/fastlane/issues/22182
